File: Java/bruch_klasse/src/main/java/blogic/Main/Bruch.java
package blogic.Main;

public class Bruch {
    
	private int zaehler = 0;
	private int nenner = 1;


	private void kuerzen(){
		int temp;
		int a = this.zaehler;
		int b = this.nenner;
		while (b != 0) {
			temp = a % b;
			a = b;
			b = temp;
		}
		this.zaehler /= a;
		this.nenner /= a;
	}


	public final boolean setWert(int z, int n){
		if (this.nenner != 0) {
			this.zaehler = z;
			this.nenner = n;
			kuerzen();
			return true;
		}
		return false;
	}

	public double getWert() {
		return (double)this.zaehler / this.nenner;
	}

	public void printWert() {
            double a = (double)this.zaehler / this.nenner;
		System.out.println(a);
	}
        
        public Bruch(int z, int n) {
            setWert(z, n);
	}
}

File: C++/Bruch_Klasse/Bruch_Klasse/Bruch.cpp
#include "Bruch.h"
#include <iostream>


	void Bruch::kuerzen(){
		int temp;
		unsigned int a = Bruch::zaehler;
		unsigned int b = Bruch::nenner;
		while (b != 0) {
			temp = a % b;
			a = b;
			b = temp;
		}
		Bruch::zaehler /= a;
		Bruch::nenner /= a;
	}

	Bruch::Bruch(unsigned int z, unsigned int n) {
		setWert(z, n);
	}

	bool Bruch::setWert(unsigned int z, unsigned int n) {
		if (nenner != 0) {
			Bruch::zaehler = z;
			Bruch::nenner = n;
			kuerzen();
			return true;
		}
		return false;
	}

	double Bruch::getWert() {
		return double(zaehler) / nenner;
	}

	void Bruch::printWert() {
		std::cout << Bruch::getWert() << std::endl;
	}